File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> helper(vector<int> list, TreeNode* root) {
        if (root == nullptr) { return list; }
        else {
            list.push_back(root->val);
        }
        
        list = helper(list, root->left);
        list = helper(list, root->right);
        return list;
    }
    
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> list;
        list = helper(list, root);
        return list;
    }
};

File: LeetCode/Easy/0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def helper(self, result: List[TreeNode], node: Optional[TreeNode]) -> None:
        if node == None:
            return
        else:
            result.append(node.val)
            
        self.helper(result, node.left)
        self.helper(result, node.right)
        return
        
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.helper(result, root)
        return result